ArduCopter/system.cpp

```diff
@@ -256,12 +256,20 @@ void Copter::update_dynamic_notch()
 
 #if RPM_ENABLED == ENABLED
         case HarmonicNotchDynamicMode::UpdateRPM: // rpm sensor based tracking
-            float rpm;
-            if (rpm_sensor.get_rpm(0, rpm)) {
-                // set the harmonic notch filter frequency from the main rotor rpm
-                ins.update_harmonic_notch_freq_hz(MAX(ref_freq, rpm * ref / 60.0f));
+            if (ins.has_harmonic_option(HarmonicNotchFilterParams::Options::DynamicHarmonic)) {
+                float notches[INS_MAX_NOTCHES];
+                const uint8_t num_notches = rpm_sensor.get_motor_frequencies_hz(ins.get_num_gyro_dynamic_notches(), notches);
+
+                for (uint8_t i = 0; i < num_notches; i++) {
+                    notches[i] =  MAX(ref_freq, notches[i]);
```

Comment:
You need to multiply by ref here. This was always important on heli

Reply:
Sorry, which specific line needs to be multiplied by ref? the notches[i] line? I don't see it in the blheli telem version so I'm not clear on what the ref does. Thanks!